src/gui/tray/activitydata.cpp

```diff
@@ -50,6 +50,22 @@ OCC::Activity Activity::fromActivityJson(const QJsonObject &json, const AccountP
 {
     const auto activityUser = json.value(QStringLiteral("user")).toString();
 
+    const auto stringToUrl = [&] (const QString link) {
+        auto url = QUrl::fromUserInput(link);
+        //QUrl::fromUserInput
+        if (url.isValid()) {
```

Comment:
You could do an early out `if (url.isValid()) { return url; }` and reduce amount of nested `if`